src/frontends/ir/tests/frontend_test_mmap.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_test_utils/file_utils.hpp"
+#include "frontend_test.hpp"
+#include "openvino/opsets/opset1.hpp"
+
+class IRFrontendMMapTestsAdvanced : public ::testing::TestWithParam<bool>, public IRFrontendTestsImpl {
+protected:
+    size_t binsize;
+
+    void SetUp() override {
+        size_t CONST_SIZE = 10000000; /*~ 77 MB for f64 to guarantee size is more than size of mapped libraries*/
+        auto parameter = std::make_shared<ov::opset1::Parameter>(ov::element::f64, ov::Shape{CONST_SIZE});
+        parameter->set_friendly_name("input");
+        auto constant = std::make_shared<ov::opset1::Constant>(ov::element::f64,
+                                                               ov::Shape{CONST_SIZE},
+                                                               std::vector<double>(CONST_SIZE, 0));
+        constant->set_friendly_name("value1");
+        auto add = std::make_shared<ov::opset1::Add>(parameter, constant);
+        add->set_friendly_name("Add");
+        auto result = std::make_shared<ov::opset1::Result>(add);
+        result->set_friendly_name("output");
+        auto model = std::make_shared<ov::Model>(ov::NodeVector{result}, ov::ParameterVector{parameter});
+
+        auto filePrefix = CommonTestUtils::generateTestFilePrefix();
+        xmlFileName = filePrefix + "_IrFrontendTestModel.xml";
+        binFileName = filePrefix + "_IrFrontendTestModel.bin";
+        ov::serialize(model, xmlFileName, binFileName);
+        binsize = CommonTestUtils::fileSize(binFileName) / 1024;
+    }
+
+    void TearDown() override {
+        RemoveTemporalFiles();
+    }
+};
+
+TEST_P(IRFrontendMMapTestsAdvanced, core_read_and_compile_model) {
+    // Test checks that with `mmap` enabled .bin file
+    // isn't read into RAM on `read_model` stage and
+    // maps into RAM on `compile_model` stage
+
+    bool is_mmap = GetParam();
+
+    // trigger plugin loading in order to map libraries and not affect memory values during compilation
+    core.get_versions("CPU");
+    core.set_property(ov::enable_mmap(GetParam()));
+
+    auto rss_init = CommonTestUtils::getVmRSSInKB();
+    auto model = core.read_model(xmlFileName, binFileName);
+    auto rss_read = CommonTestUtils::getVmRSSInKB();
+
+    // RAM should increase on at least 90% of .bin size. 10% is a proposed error that cover file system cache
+    size_t bin_in_RAM = binsize * 0.9;
```

Comment:
As discussed:
1. Test is run in a separate process, so memory behavior became independent from other tests
2. `compile_model` is removed, so `RssFile` metric is also gone (it shows how which mapped memory is loaded)

Anyway, I decided to leave RSS delta instead of full RSS metric because makes test not dependent from other components (libraries size, Core etc.). If some prereading of .bin will be implemented out of `read_model` scope, it will be found by the test with `enable_mmap(false)`